e2e-tests/expose-sharded/run

```diff
@@ -0,0 +1,232 @@
+		desc "Actual values rs $rs0_0_endpoint_actual and cfg $cfg_0_endpoint_actual do not match expected rs $rs0_0_endpoint:27017 and cfg $cfg_0_endpoint:27017"
+		exit 1
+	fi
+
+}
+
+function expose_cluster() {
+	expose_type=$1
+	expose_status=${2:-true}
+
+	kubectl_bin patch psmdb ${cluster} --type=json --patch '[
+          {
+              "op": "replace",
+              "path": "/spec/replsets/0/expose",
+        "value": {
+          "enabled": '$expose_status',
+          "exposeType" : "'"${expose_type}"'"
+        }
+          },
+        {
+              "op": "replace",
+              "path": "/spec/sharding/mongos/expose",
+        "value": {
+          "enabled": '${expose_status}',
+          "exposeType" : "'"${expose_type}"'"
+        }
+          },
+          {
+              "op": "replace",
+              "path": "/spec/sharding/configsvrReplSet/expose",
+        "value": {
+          "enabled": '${expose_status}',
+          "exposeType" : "'"${expose_type}"'"
+        }
+          }]'
+}
+function main() {
+
+	create_infra "$namespace"
+	desc 'create first PSMDB cluster'
+	cluster="some-name"
+
+	kubectl_bin apply \
+		-f "$conf_dir/secrets.yml" \
+		-f "$conf_dir/client.yml"
+
+	apply_s3_storage_secrets
+	if version_gt "1.19" && [ $EKS -ne 1 ]; then
+		cat "$conf_dir/container-rc.yaml" | $sed 's/docker/runc/g' | kubectl_bin apply -f -
+	else
+		kubectl_bin apply -f "$conf_dir/container-rc.yaml"
+	fi
+
+	apply_cluster "$test_dir/conf/$cluster-rs0.yml"
+	desc 'check if all 3 Pods started'
+	wait_for_running $cluster-rs0 3
+	wait_for_running $cluster-cfg 3 "false"
+	wait_for_running $cluster-mongos 3
+
+	sleep 10
+
+	desc 'check if service and statefulset created with expected config'
+	compare_kubectl statefulset/$cluster-rs0
+	compare_kubectl statefulset/$cluster-cfg
+	compare_kubectl statefulset/$cluster-mongos ""
+
+	desc 'write data, read from all'
+	run_mongos \
+		'db.createUser({user:"myApp",pwd:"myPass",roles:[{db:"myApp",role:"readWrite"}]})' \
+		"userAdmin:userAdmin123456@$cluster-mongos.$namespace"
+	run_mongo 'db.createUser({user: "myApp", pwd: "myPass", roles: [{ db: "myApp", role: "readWrite" }]})' \
+		"userAdmin:userAdmin123456@$cluster-rs0-0.$cluster-rs0.${namespace}" "mongodb"
+
+	run_mongos \
+		'sh.enableSharding("myApp","rs0")' \
+		"clusterAdmin:clusterAdmin123456@$cluster-mongos.$namespace"
+	run_mongos \
+		'use myApp\n db.test.insert({ x: 100500 })' \
+		"myApp:myPass@$cluster-mongos.$namespace"
+	sleep 10 # for minikube
+	compare_mongos_cmd "find" "myApp:myPass@$cluster-mongos.$namespace"
+
+	desc 'Unexposed -> Exposed, ClusterIP'
+	expose_cluster "ClusterIP"
+
+	wait_for_running $cluster-rs0 3
+	wait_for_running $cluster-cfg 3 "false"
+	wait_for_running $cluster-mongos 3
+	run_mongos \
+		'use myApp\n db.test.insert({ x: 100501 })' \
+		"myApp:myPass@$cluster-mongos.$namespace"
+
+	sleep 10 # for minikube
```

Comment:
I propose to check if we run on minikube and only in that case to do sleep. Maybe like this: https://github.com/percona/percona-server-mysql-operator/blob/main/e2e-tests/vars.sh#L33-L35